src/Core/src/Hosting/AppHostBuilderExtensions.cs

```diff
@@ -98,36 +105,15 @@ public static IAppHostBuilder UseMauiServiceProviderFactory(this IAppHostBuilder
 
 		public static IAppHostBuilder EnableHotReload(this IAppHostBuilder builder, string? ideIp = null, int idePort = 9988)
 		{
-			builder.ConfigureMauiHandlers((context, handlersCollection) =>
+			builder.ConfigureServices<HotReloadBuilder>(hotReload =>
```

Comment:
I switched hot reload to use the builder patter as this gives us better control as to when things are actually run.